examples/demo10.yaml

```diff
@@ -216,3 +219,89 @@ output :
         # main image.
         signal_to_noise : 1000
 
+        # Offset the PSF images randomly within the central pixel to better sample the
+        # sub-pixel surface brightness profile.  Note: as with the output.psf.shift parameter
+        # (cf. demo5.yaml), the output field may also take the value 'galaxy', which would mean
+        # to apply the same offset as was applied to the galaxy.  In this case, there is no
+        # such offset, so it wouldn't apply here, but it's an option if you want to match the
+        # offsets of the PSF and galaxy.
+        offset:
+            type : XY
+            x : { type: Random, min: -0.5, max: 0.5 }
+            y : { type: Random, min: -0.5, max: 0.5 }
+
+
+    # Make a truth catalog as a binary table in hdu 2
+    truth :
+        hdu : 2
+
+        columns :
+            gal_num : obj_num
+
+            # We saw in demo9 that position values in the output catalog get converted into two
+            # float values, and the column names get '.x' and '.y' appended to the given name.
+            # If you want, you can instead specify the x and y column names directly, but you
+            # need to do a little calculation to get the values, since these aren't natively
+            # stored anywhere.  To enable this, truth columns can be calculated quantities,
+            # just like any other field in the configuration.  However, only Eval and Current
+            # types are allowed.
+            x_image : { type : Eval, str : 'image_pos.x' }
+            y_image : { type : Eval, str : 'image_pos.y' }
+
+            # Similarly, shear values append '.g1' and '.g2' to the name and output a column
+            # for each of the two components.  However, this isn't necessarily what you want.
+            # E.g. you might want to catalog the e1,e2 definitions of the shape rather than g1,g2.
+            psf_e1 :
+                type : Eval
+                str : 'ellip.e1'
+                # Reminder g is the Eval code for Shear types. (s = str was already taken)
+                gellip : { type : Current, key : 'psf.ellip' }
+            psf_e2 :
+                type : Eval
+                str : 'ellip.e2'
+                gellip : { type : Current, key : 'psf.ellip' }
+
+            psf_fwhm : psf.fwhm
+
+            cosmos_id : gal.first.id
+
+            # Even though we accessed the catalog by id, we can still get the index number
+            # in the catalog via the gal.index attribute.
+            cosmos_index :
+                type : Eval
+                str : 'gal.index'
+                # There is a special Eval code we haven't mentioned yet.  If you precede a
+                # variable name with x, then that means it's an object, rather than a value.
+                # So in this case, get the index from the attribute of the object, gal.first.
+                xgal : { type : Current, key : 'gal.first' }
```

Comment:
Good point on the overloaded name here.  Maybe calling the variable `first_gal` would be clearer?

And yes, the `x` code is a bit different from the rest, since it refers to a GSObject, which is what I meant by "object" here.  I'll add a bit more to try to make that clearer.